tests/test_protocol.py

```diff
@@ -611,6 +611,18 @@ def test_method(project: Project) -> ReturnValue[Project]:  # NOQA
 
             return ReturnValue(response=new_project)
 
+        @protocol.typedmethod(path="/test2", operation="POST", client_types=["api"], envelope_key="method")
```

Comment:
Shouldn't we test the envelope feature for the non-typed (protocol.method) as well.